src/commands/mod.rs

```diff
@@ -490,6 +494,105 @@ impl DaemonControl {
             Err(BitcoinError::Broadcast(e)) => Err(CommandError::TxBroadcast(e)),
         }
     }
+
+    /// gethistory retrieves a limited list of events which occured between two given dates.
+    pub fn gethistory(&self, start: u32, end: u32, limit: u64) -> GetHistoryResult {
+        let mut db_conn = self.db.connection();
+        let coins = db_conn.list_updated_coins(start, end, limit);
+
+        // All the spends occuring in the bound.
+        let mut spends: HashMap<bitcoin::Txid, Vec<&Coin>> = HashMap::with_capacity(coins.len());
+
+        // Preparatory work to populate spends and all_spend_txids.
+        for coin in &coins {
+            if let Some(txid) = coin.spend_txid {
+                let time = coin.spend_block.expect("Coin is spent").time;
```

Comment:
You can't assume a Spend was mined on the presence of its txid. This will trigger a crash when there is an unconfirmed Spend.